packages/docgen/src/get-export-entries.js

```diff
@@ -0,0 +1,97 @@
+/**
+ * External dependencies
+ */
+const { get } = require( 'lodash' );
+
+/**
+ * Returns the export entry records of the given export statement.
+ * Unlike [the standard](http://www.ecma-international.org/ecma-262/9.0/#exportentry-record),
+ * the `importName` and the `localName` are merged together.
+ *
+ * @param {Object} token Espree node representing an export.
+ *
+ * @return {Array} Exported entry records. Example:
+ * [ {
+ *    localName: 'localName',
+ *    exportName: 'exportedName',
+ *    module: null,
+ *    lineStart: 2,
+ *    lineEnd: 3,
+ * } ]
+ */
+module.exports = function( token ) {
+	if ( token.type === 'ExportDefaultDeclaration' ) {
+		const getLocalName = ( t ) => {
+			let name;
+			switch ( t.declaration.type ) {
+				case 'Identifier':
+					name = t.declaration.name;
+					break;
+				case 'AssignmentExpression':
+					name = t.declaration.left.name;
+					break;
+				//case 'FunctionDeclaration'
```

Comment:
Should have been removed?

Reply:
From memory, I think it documents the cases that fall-through.

Reply:
> From memory, I think it documents the cases that fall-through.

It could also be explicit, no?

```js
switch ( t.declaration.type ) {
	// ...

	case 'FunctionDeclaration':
	case 'ClassDeclaration':
	default:
		// ...
}
```

_(If it were a web bundle, we may have concern about the "unnecessary" bytes added, but it's not in this instance)_

I'd just worry that this might not be obvious to the future maintainer without at least a hint (inline comment) to explain the intention.